src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -0,0 +1,555 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    internal sealed class ControlFlowGraphBuilder : OperationCloner
+    {
+        private readonly BasicBlock _entry = new BasicBlock(BasicBlockKind.Entry);
+        private readonly BasicBlock _exit = new BasicBlock(BasicBlockKind.Exit);
+        private ArrayBuilder<BasicBlock> _blocks;
+        private BasicBlock _currentBasicBlock;
+        private IOperation _currentStatement;
+        private ArrayBuilder<IOperation> _evalStack;
+        private int _availableCaptureId;
```

Comment:
>_availableCaptureId [](start = 20, length = 19)

Nit: Consider making default value explicit.
Related: does the public API `IFlowCaptureOperation.Id` specify how identifiers are created or assigned? Could we use `uint` to exclude negative integers? I wonder if we should randomize them in any way to avoid dependencies being taken...

Reply:
Added a comment

---
In reply to: [162176797](https://github.com/dotnet/roslyn/pull/24263#discussion_r162176797) [](ancestors = 162176797)